docs/flow/config-language/components.md

```diff
@@ -6,3 +6,157 @@ weight: 300
 ---
 
 # Components
+Components are the defining feature of Grafana Agent Flow. They are autonomous
+pieces of business logic that perform a single task (like retrieving secrets or
+collecting Prometheus metrics) and are wired together to form programmable
+pipelines of telemetry data.
+
+Under the hood, components are orchestrated via the [_component
+controller_]({{< relref "../concepts/component_controller.md" >}}), who is
+responsible for scheduling them, reporting their health and debug status, and
+continuously re-evaluating their inputs and outputs.
+
+Each top-level River _block_ will instantiate a new component. A component
+consists of its name (which describes what the component is responsible for) as
+well as an optional user-specified label.
+
+You can see a list of all available components [here]({{< relref "../components/_index.md" >}}).
+Each component features a complete reference page, so getting a component to
+work for you should be as easy as reading its documentation and copy/pasting
+from an example.
+
+## Arguments and Exports
+Most user interactions with components will center around two basic concepts;
+_Arguments_ and _Exports_.
+
+* _Arguments_ work much like function arguments in your favorite programming
+ language. They can be any number of attributes or nested unlabeled blocks and
+are used to configure the component's behavior. Most components will feature
+both required and optional arguments. Any arguments that are not provided will
+take on their default values.
+
+* _Exports_ are zero or more named return values that can be referred to by
+ other components. There is no limit to what an export might be; it can take on
+any River value or Go type (eg. channels, interfaces).
+
+Each component's documentation page contains the list of all available
+Arguments and resulting Exports, along their underlying type. Users can
+configure Arguments either using a constant value or an
+[_expression_]({{< relref "./expressions/_index.md" >}}) to continuously
+re-evaluate values who are dynamic in nature.
```

Comment:
I think the concepts page already links to the reference list of components and explains how reevaluation works, doesn't it? We should probably remove this paragraph and leave it to the concept-level explanation.